WooCommerce/Classes/ViewRelated/Dashboard/Reviews/ReviewsDashboardCardViewModel.swift

```diff
@@ -86,105 +117,162 @@ private extension ReviewsDashboardCardViewModel {
+            updateData()
         } catch {
             ServiceLocator.crashLogging.logError(error)
         }
     }
 
-    func updateProductsResultsControllerPredicate(with productIDs: [Int64]) {
-        let predicates = NSCompoundPredicate(andPredicateWithSubpredicates: [sitePredicate(),
-                                                                             NSPredicate(format: "productID IN %@", productIDs)])
-        productsResultsController.predicate = predicates
+    func configureProductsResultsController() {
+        productsResultsController.onDidChangeContent = { [weak self] in
+            guard let self else { return }
+            self.updateData()
+        }
+        productsResultsController.onDidResetContent = { [weak self] in
+            guard let self else { return }
+            self.updateData()
+        }
+
+        /// Note: Intentionally not doing performFetch() here, as it has to wait for available
+        /// productIDs from productReviewsResultsController() for the results to be correct.
+    }
+
+    func configureNotificationsResultsController() {
+        notificationsResultsController.predicate = notificationsPredicate
+
+        notificationsResultsController.onDidChangeContent = { [weak self] in
+            guard let self else { return }
+            self.updateData()
+        }
+        notificationsResultsController.onDidResetContent = { [weak self] in
+            guard let self else { return }
+            self.updateData()
+        }
+
+        do {
+            try notificationsResultsController.performFetch()
+            updateData()
+        } catch {
+            ServiceLocator.crashLogging.logError(error)
+        }
     }
 }
 
 // MARK: - Private helpers
 private extension ReviewsDashboardCardViewModel {
-    @MainActor
-    func loadReviews() async throws {
-        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
-            stores.dispatch(ProductReviewAction.synchronizeProductReviews(siteID: siteID,
-                                                                          pageNumber: 1,
-                                                                          pageSize: Constants.numberOfItems) { result in
-                switch result {
-                case .success:
-                    // Ignoring the result from remote as we're using storage as the single source of truth
-                    continuation.resume()
-                case .failure(let error):
-                    continuation.resume(throwing: error)
+    /// Populates data from the local storage for the three main data (reviews, products, notifications).
+    ///
+    func updateData() {
+        let localReviews = reviews.prefix(Constants.numberOfItems)
+
+        // We can show partial review content as long as there are reviews found in storage.
+        if localReviews.isEmpty == false {
+            showLoadingAnimation = false
+        }
```

Comment:
We should follow the "separation of concern" principle here. It's better to not update the UI while working on data. Instead, you can observe changes in `data` and update `showLoadingAnimation` to false when it's not empty.

Reply:
TIL for me, updated in https://github.com/woocommerce/woocommerce-ios/pull/12769/commits/4ba6ea01b764696f6722e197be376018d7e6e4b1